pseudomonas_analysis/nbconverted/2_identify_generic_genes_pathways.py

```diff
@@ -0,0 +1,337 @@
+# PA4527 maps to pilC and still frameshift type 4 fimbrial biogenesis protein PilC (putative pseudogene)
+gene_name_mapping = gene_name_mapping[~gene_name_mapping.index.duplicated(keep=False)]
+
+
+# In[20]:
+
+
+# Add gene names
+#gene_name_mapping_dict = gene_name_mapping.to_dict()
+summary_gene_ranks['Gene Name'] = summary_gene_ranks['Gene ID'].map(gene_name_mapping["Name"])
+summary_gene_ranks.head()
+
+
+# In[21]:
+
+
+summary_gene_ranks.to_csv(
+    gene_summary_file, sep='\t')
+
+
+# ### GSEA 
+# **Goal:** To detect modest but coordinated changes in prespecified sets of related genes (i.e. those genes in the same pathway or share the same GO term).
+# 
+# 1. Ranks all genes based using DE association statistics. In this case we used the p-value scores to rank genes. logFC returned error -- need to look into this.
+# 2. An enrichment score (ES) is defined as the maximum distance from the middle of the ranked list. Thus, the enrichment score indicates whether the genes contained in a gene set are clustered towards the beginning or the end of the ranked list (indicating a correlation with change in expression). 
+# 3. Estimate the statistical significance of the ES by a phenotypic-based permutation test in order to produce a null distribution for the ES( i.e. scores based on permuted phenotype)
+
+# ### Rank pathways 
+
+# ### Pathway summary table
+
+# ### Compare gene rankings
+
+# In[22]:
+
+
+if compare_genes:
+    # Get generic genes identified by Crow et. al.
+    DE_prior_file = params['reference_gene_file']
+    ref_gene_col = params['reference_gene_name_col']
+    ref_rank_col = params['reference_rank_col']
+    
+    # Merge our ranking and reference ranking
+    shared_gene_rank_df = process.merge_ranks_to_compare(
+        summary_gene_ranks,
+        DE_prior_file,
+        ref_gene_col,
+        ref_rank_col)
+    
+    if max(shared_gene_rank_df["Rank (simulated)"]) != max(shared_gene_rank_df[ref_rank_col]):
+        shared_gene_rank_scaled_df = process.scale_reference_ranking(shared_gene_rank_df, ref_rank_col)
+        
+    # Get correlation
+    r, p, ci_high, ci_low = calc.spearman_ci(0.95,
+                                             shared_gene_rank_scaled_df,
+                                             1000)
+    print(r, p, ci_high, ci_low)
+    
+    # Plot our ranking vs published ranking
+    fig_file = os.path.join(
+        local_dir, 
+        "gene_ranking_"+col_to_rank+".svg")
+
+    fig = sns.jointplot(data=shared_gene_rank_scaled_df,
+                        x='Rank (simulated)',
+                        y=ref_rank_col,
+                        kind='hex',
+                        marginal_kws={'color':'white'})
+    fig.set_axis_labels("Our preliminary method", "Reference Name", fontsize=14)
+
+    fig.savefig(fig_file,
+                format='svg',
+                bbox_inches="tight",
+                transparent=True,
+                pad_inches=0,
+                dpi=300,)
```

Comment:
In this case, there is nothing plotted because we don't have a reference set of genes that we are comparing against. I kept it here to be consistent with the recount2 analysis so if a user tries to run their own analysis and make a copy of the notebook they didn't have to change cells too much...not sure what the best tactic is for this.